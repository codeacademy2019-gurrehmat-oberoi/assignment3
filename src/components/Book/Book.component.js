import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Book.css';

const imagePath = require('../../media/cover_image.jpeg');
const heartIcon = require('../../media/Icons/heart-red.svg');

class Book extends Component {
  render() {
    const { bookData } = this.props;
    return (
      <div className="book">
        <img src={imagePath} alt="Book cover" className="bookCover" />
        <div className="bookTitle">
          {bookData.title}
        </div>
        <div className="bookAppreciation">
          <div className="bookRating">{bookData.rating}</div>
          <div className="bookLikeButton">
            <img src={heartIcon} alt="like book" />
          </div>
        </div>
      </div>
    );
  }
}

Book.defaultProps = {
  bookData: {
    title: 'Harry Potter and The Sorcerers Stone',
    rating: 4.5,
  },
};

Book.propTypes = {
  bookData: PropTypes.object,
};

export default Book;
