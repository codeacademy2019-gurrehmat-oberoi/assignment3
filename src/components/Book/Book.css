.book {
  display: flex;
  flex-direction: column;
  width: 30%;
  min-width: 200px;
  align-items: stretch;
  font-size: 30px;
}

.book .bookCover {
  width: 100%;
}

.book .bookTitle {
  text-align: center;
}

.book .bookAppreciation{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

}

.book .bookAppreciation .bookLikeButton {
  width: 25px;
}