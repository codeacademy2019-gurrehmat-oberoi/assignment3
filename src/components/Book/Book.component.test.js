import renderer from 'react-test-renderer';
import React from 'react';
import Book from './Book.component';

describe('Book', () => {
  it('should render a book', () => {
    const tree = renderer.create(<Book />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
