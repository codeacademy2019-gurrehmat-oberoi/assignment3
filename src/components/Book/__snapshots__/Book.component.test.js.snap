// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render a book 1`] = `
<div
  className="book"
>
  <img
    alt="Book cover"
    className="bookCover"
    src="cover_image.jpeg"
  />
  <div
    className="bookTitle"
  >
    Harry Potter and The Sorcerers Stone
  </div>
  <div
    className="bookAppreciation"
  >
    <div
      className="bookRating"
    >
      4.5
    </div>
    <div
      className="bookLikeButton"
    >
      <img
        alt="like book"
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "heart-red.svg",
          }
        }
      />
    </div>
  </div>
</div>
`;
